### ISSUE TITLE
[Bug]: Scroll of Essence Used Up in Non-essence Mining

### ISSUE THREAD

**Issue by SevnirOG** [MelvorIdle/melvoridle.github.io#4133]
### Describe the bug

TL;DR Scrolls of Essence are used up while mining non-essence ores, but do not give any essence, so they are just wasted.
I had Scroll of Essence equipped for mining essence, then switched to mining runite ore without unequipping the scrolls. The Scrolls of Essence were used up for each runite ore (approx 20k scrolls) without giving any rune essence. I wasn't expecting to get essence while mining runite, but the scrolls should not have been used up either. 

### Reproduction Steps

1. Equip Scrolls of Essence 
2. Mine essence then switch to mining runite ore
3. Watch Scrolls of Essence disappear with each runite ore

### Expected behaviour

Scrolls of Essence shouldn't decrease when mining non-essence ore

### Which platforms are you experiencing this bug on?

- [X] Chrome
- [ ] Firefox
- [ ] Edge
- [ ] Safari
- [ ] Steam
- [ ] Epic Games
- [ ] Mobile App (iOS)
- [X] Mobile App (Android)
- [ ] Mobile App (Android Beta)
- [ ] Other (Please Specify)

### Which expansions are purchased and active at the time of the bug?

- [X] Throne of the Herald
- [X] Atlas of Discovery

### Which version of the game are you experiencing this bug on?

v1.2.2

### Game Subversion

Unknown

### Are you using any mods?

No
